jupyterhub_traefik_proxy/etcd.py

```diff
@@ -18,30 +18,44 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+from concurrent.futures import ThreadPoolExecutor
 import json
 from urllib.parse import urlparse
-from traitlets import Any, default, Unicode
 
-from aioetcd3.client import client
-from aioetcd3 import transaction
-from aioetcd3.kv import KV
-from aioetcd3.help import range_prefix
+import etcd3
+from tornado.concurrent import run_on_executor
+from traitlets import Any, default, Unicode
 
+from jupyterhub.utils import maybe_future
 from . import traefik_utils
 from jupyterhub_traefik_proxy import TraefikProxy
 
 
 class TraefikEtcdProxy(TraefikProxy):
     """JupyterHub Proxy implementation using traefik and etcd"""
 
+    executor = Any()
+
+    @default("executor")
+    def _default_executor(self):
+        return ThreadPoolExecutor(1)
+
     etcd_client = Any()
 
+    etcd_password = Unicode(config=True, help="""The password for etcd login""")
+
     @default("etcd_client")
     def _default_client(self):
         etcd_service = urlparse(self.etcd_url)
-        return client(
-            endpoint=str(etcd_service.hostname) + ":" + str(etcd_service.port)
-        )
+        if self.etcd_password:
+            return etcd3.client(
+                host=str(etcd_service.hostname),
+                port=etcd_service.port,
+                user="root",
```

Comment:
`etcd_username` should probably be a configurable trait like etcd_password. `root` can be the default, if desired.